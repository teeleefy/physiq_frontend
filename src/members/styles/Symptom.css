.Symptom-card {
    height: 25vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Symptom-label {
    font-size: 1.5rem;
}

.Symptom-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Symptoms-navlink {
    text-decoration: none;
    color: white;
}

.Symptom-date-input {
    width: 20vw;
}