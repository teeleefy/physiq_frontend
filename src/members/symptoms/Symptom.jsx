import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
  import '../styles/Symptom.css'


function Symptom({symptom, memberId}){

    function renderStartDate(){
        if(symptom.startDate){
            return (
                <>
                 <p><b>Start Date:</b> {symptom.startDate}</p>
                </>
            )
        }
    }

    function renderEndDate(){
        if(symptom.endDate){
            return (
                <>
                 <p><b>End Date:</b> {symptom.endDate}</p>
                </>
            )
        }
    }

    function renderNotes(){
        if(symptom.notes){
            return (
                <>
                 <p><b>Notes:</b> {symptom.notes}</p>
                </>
            )
        }
    }

    return(
        <>
            <ListGroupItem >
                <Card className="Symptom-card">
                    <CardBody>
                        <CardText>
                            <p><b>Symptom Name:</b> <NavLink className="Symptom-name" to={`/member/${memberId}/symptoms/${symptom.id}`} key={symptom.id}>
                                {symptom.name}
                            </NavLink></p>
                            {renderStartDate()}
                            {renderEndDate()}
                            {renderNotes()}
                        </CardText>
                    </CardBody>
                </Card>
            </ListGroupItem>
        </>
     )
}

export default Symptom;