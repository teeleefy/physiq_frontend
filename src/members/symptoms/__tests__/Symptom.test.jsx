import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Symptom from '../Symptom'

let symptom={name: "Headache", startDate: "2024-07-20", endDate: "2024-08-01", notes: "Moderate to severe. On and off every day. Worse at night."}
let memberId=1000;

describe('Symptom', () => {
  it('renders the Symptom component', () => {
    renderWithRouter(<Symptom symptom={symptom} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Symptom component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Symptom symptom={symptom} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});