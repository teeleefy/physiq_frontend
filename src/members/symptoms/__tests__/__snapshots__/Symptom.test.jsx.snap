// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Symptom-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Symptom Name:
          </b>
           
          <a
            class="Symptom-name"
            href="/member/1000/symptoms/undefined"
          >
            Headache
          </a>
        </p>
        <p>
          <b>
            Start Date:
          </b>
           2024-07-20
        </p>
        <p>
          <b>
            End Date:
          </b>
           2024-08-01
        </p>
        <p>
          <b>
            Notes:
          </b>
           Moderate to severe. On and off every day. Worse at night.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
